File: components/vsod/constants/headers.js
// Copyright (c) 2026 Lateralus Labs, LLC.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { _HEADERS } from './shared.js';

/**
 * VSOHeaders - X-VSO-* HTTP header name constants.
 * Canonical values loaded from shared/constants/headers.json.
 * That file is the single source of truth shared across g8ee, g8eo, and VSOD.
 *
 * All internal cluster-to-cluster HTTP requests use these header names to
 * propagate VSOHttpContext (session, identity, business context) between
 * components without re-authentication.
 *
 * @type {{ [key: string]: string }}
 */
export const VSOHeaders = Object.freeze({
    WEB_SESSION_ID:    _HEADERS['x-vso.session-id'],
    USER_ID:           _HEADERS['x-vso.user-id'],
    ORGANIZATION_ID:   _HEADERS['x-vso.organization-id'],
    CASE_ID:           _HEADERS['x-vso.case-id'],
    INVESTIGATION_ID:  _HEADERS['x-vso.investigation-id'],
    TASK_ID:           _HEADERS['x-vso.task-id'],
    SOURCE_COMPONENT:  _HEADERS['x-vso.source-component'],
    BOUND_OPERATORS:   _HEADERS['x-vso.bound-operators'],
    EXECUTION_ID:      _HEADERS['x-vso.execution-id'],
    NEW_CASE:          _HEADERS['x-vso.new-case'],
    SERVICE:           _HEADERS['x-vso.service'],
    CLIENT:            _HEADERS['x-vso.client'],
    OPERATOR_STATUS:   _HEADERS['x-vso.operator-status'],
});

export const HTTP_REQUESTED_WITH_HEADER        = _HEADERS['http.requested-with'];
export const HTTP_VSO_SERVICE_HEADER           = _HEADERS['x-vso.service'];
export const HTTP_VSO_CLIENT_HEADER            = _HEADERS['x-vso.client'];
export const HTTP_VSO_OPERATOR_STATUS_HEADER   = _HEADERS['x-vso.operator-status'];
export const HTTP_CACHE_CONTROL_HEADER         = _HEADERS['http.cache-control'];
export const HTTP_PRAGMA_HEADER                = _HEADERS['http.pragma'];
export const HTTP_COOKIE_HEADER                = _HEADERS['http.cookie'];
export const HTTP_SET_COOKIE_HEADER            = _HEADERS['http.set-cookie'];
export const HTTP_LAST_EVENT_ID_HEADER         = _HEADERS['http.last-event-id'];
export const HTTP_ACCESS_CONTROL_REQUEST_HEADERS = _HEADERS['http.access-control-req-headers'];
export const HTTP_ACCESS_CONTROL_REQUEST_METHOD  = _HEADERS['http.access-control-req-method'];
export const HTTP_ACCESS_CONTROL_ALLOW_ORIGIN    = _HEADERS['http.access-control-allow-origin'];
export const HTTP_ACCESS_CONTROL_ALLOW_CREDENTIALS = _HEADERS['http.access-control-allow-creds'];
export const HTTP_CONTENT_TYPE_HEADER          = _HEADERS['http.content-type'];
export const HTTP_API_KEY_HEADER               = _HEADERS['http.api-key'];
export const WEB_SESSION_ID_HEADER             = _HEADERS['x-vso.session-id'];
export const HTTP_INTERNAL_AUTH_HEADER          = _HEADERS['http.x-internal-auth'];
export const HTTP_X_FORWARDED_HOST_HEADER      = _HEADERS['http.x-forwarded-host'];
export const HTTP_X_FORWARDED_PROTO_HEADER     = _HEADERS['http.x-forwarded-proto'];

File: components/g8ee/app/constants/headers.py
# Copyright (c) 2026 Lateralus Labs, LLC.
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from enum import Enum

HTTP_ACCEL_BUFFERING_HEADER = "X-Accel-Buffering"
HTTP_ACCEPT_HEADER = "Accept"
HTTP_ACCEPT_LANGUAGE_HEADER = "Accept-Language"
HTTP_ACCESS_CONTROL_ALLOW_CREDENTIALS = "Access-Control-Allow-Credentials"
HTTP_ACCESS_CONTROL_ALLOW_ORIGIN = "Access-Control-Allow-Origin"
HTTP_ACCESS_CONTROL_REQUEST_HEADERS = "Access-Control-Request-Headers"
HTTP_ACCESS_CONTROL_REQUEST_METHOD = "Access-Control-Request-Method"
HTTP_API_KEY_HEADER = "X-API-Key"
HTTP_AUTHORIZATION_HEADER = "Authorization"
HTTP_BEARER_PREFIX = "Bearer"
HTTP_CACHE_CONTROL_HEADER = "Cache-Control"
HTTP_CONTENT_LANGUAGE_HEADER = "Content-Language"
HTTP_CONTENT_TYPE_HEADER = "Content-Type"
HTTP_COOKIE_HEADER = "Cookie"
HTTP_FORWARDED_FOR_HEADER = "X-Forwarded-For"
HTTP_LAST_EVENT_ID_HEADER = "Last-Event-ID"
HTTP_PRAGMA_HEADER = "Pragma"
HTTP_REQUESTED_WITH_HEADER = "X-Requested-With"
HTTP_SET_COOKIE_HEADER = "Set-Cookie"
HTTP_USER_AGENT_HEADER = "User-Agent"
HTTP_VSO_CLIENT_HEADER = "X-VSO-Client"
HTTP_VSO_OPERATOR_STATUS_HEADER = "X-VSO-Operator-Status"
HTTP_VSO_SERVICE_HEADER = "X-VSO-Service"
INTERNAL_AUTH_HEADER = "X-Internal-Auth"
PROXY_ORGANIZATION_ID_HEADER = "X-Proxy-Organization-Id"
PROXY_USER_EMAIL_HEADER = "X-Proxy-User-Email"
PROXY_USER_ID_HEADER = "X-Proxy-User-Id"
WEB_SESSION_ID_HEADER = "X-VSO-WebSession-ID"

SESSION_ID_HEADER = "X-Session-Id"
CASE_ID_HEADER = "X-VSO-Case-ID"
USER_ID_HEADER = "X-VSO-User-ID"
ORGANIZATION_ID_HEADER = "X-VSO-Organization-ID"
INVESTIGATION_ID_HEADER = "X-VSO-Investigation-ID"
TASK_ID_HEADER = "X-VSO-Task-ID"
NEW_CASE_HEADER = "X-VSO-New-Case"
BOUND_OPERATORS_HEADER = "X-VSO-Bound-Operators"
OPERATOR_ID_HEADER = "X-VSO-User-ID"
OPERATOR_SESSION_ID_HEADER = "X-VSO-WebSession-ID"
EXECUTION_ID_HEADER = "X-VSO-Execution-ID"
COMPONENT_NAME_HEADER = "X-VSO-Source-Component"

class VSOHeaders(str, Enum):
    WEB_SESSION_ID = WEB_SESSION_ID_HEADER
    USER_ID = USER_ID_HEADER
    ORGANIZATION_ID = ORGANIZATION_ID_HEADER
    CASE_ID = CASE_ID_HEADER
    INVESTIGATION_ID = INVESTIGATION_ID_HEADER
    TASK_ID = TASK_ID_HEADER
    SOURCE_COMPONENT = COMPONENT_NAME_HEADER
    BOUND_OPERATORS = BOUND_OPERATORS_HEADER
    EXECUTION_ID = EXECUTION_ID_HEADER
    NEW_CASE = NEW_CASE_HEADER
    SERVICE = HTTP_VSO_SERVICE_HEADER
    CLIENT = HTTP_VSO_CLIENT_HEADER
    OPERATOR_STATUS = HTTP_VSO_OPERATOR_STATUS_HEADER
